core/src/main/java/org/sonarsource/sonarlint/core/embedded/server/ShowIssueRequestHandler.java

```diff
@@ -0,0 +1,211 @@
+import org.sonarsource.sonarlint.core.serverapi.proto.sonarqube.ws.Issues;
+import org.sonarsource.sonarlint.core.telemetry.TelemetryServiceImpl;
+
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+
+@Named
+@Singleton
+public class ShowIssueRequestHandler extends ShowHotspotOrIssueRequestHandler implements HttpRequestHandler {
+
+  private final SonarLintClient client;
+  private final ConnectionConfigurationRepository repository;
+  private final ConfigurationServiceImpl configurationService;
+  private final ServerApiProvider serverApiProvider;
+  private final TelemetryServiceImpl telemetryService;
+
+  public ShowIssueRequestHandler(SonarLintClient client, ConnectionConfigurationRepository repository,
+    ConfigurationServiceImpl configurationService, BindingSuggestionProviderImpl bindingSuggestionProvider,
+    ServerApiProvider serverApiProvider, TelemetryServiceImpl telemetryService) {
+    super(bindingSuggestionProvider, client);
+    this.client = client;
+    this.repository = repository;
+    this.configurationService = configurationService;
+    this.serverApiProvider = serverApiProvider;
+    this.telemetryService = telemetryService;
+  }
+
+  @Override
+  public void handle(ClassicHttpRequest request, ClassicHttpResponse response, HttpContext context) throws HttpException, IOException {
+    var showIssueQuery = extractQuery(request);
+    if (!showIssueQuery.isValid()) {
+      response.setCode(HttpStatus.SC_BAD_REQUEST);
+      return;
+    }
+
+    CompletableFuture.runAsync(() -> showIssue(showIssueQuery));
+
+    response.setCode(HttpStatus.SC_OK);
+    response.setEntity(new StringEntity("OK"));
+  }
+
+  @VisibleForTesting
+  public void showIssue(ShowIssueQuery query) {
+    telemetryService.showIssueRequestReceived();
+
+    var connectionsMatchingOrigin = repository.findByUrl(query.serverUrl);
+    if (connectionsMatchingOrigin.isEmpty()) {
+      startFullBindingProcess();
+      assistCreatingConnection(query.serverUrl)
+        .thenCompose(response -> assistBinding(response.getNewConnectionId(), query.projectKey))
+        .thenAccept(response -> showIssueForScope(response.getConnectionId(), query.issueKey))
+        .whenComplete((v, e) -> endFullBindingProcess());
+    } else {
+      // we pick the first connection but this could lead to issues later if there were several matches (make the user select the right
+      // one?)
+      showIssueForConnection(connectionsMatchingOrigin.get(0).getConnectionId(), query.projectKey, query.issueKey);
+    }
+  }
+
+  private void showIssueForConnection(String connectionId, String projectKey, String issueKey) {
+    var scopes = configurationService.getConfigScopesWithBindingConfiguredTo(connectionId, projectKey);
+    if (scopes.isEmpty()) {
+      assistBinding(connectionId, projectKey)
+        .thenAccept(newBinding -> showIssueForScope(connectionId, issueKey));
+    } else {
+      showIssueForScope(connectionId, issueKey);
```

Comment:
We probably have the same problem with the show hotspot, but if the `scopes` list has more than one element, should we show the issue in all of them? I can imagine on SLI a situation where a project and a module in a different project are bound to the same connection and projectKey, we probably want to show the issues in the 2 windows

Reply:
For now, I've just added a note for future generations as we did for hotspots....